File: app/src/main/java/org/deiverbum/app/utils/ColorUtils.java
package org.deiverbum.app.utils;

import android.graphics.Color;
import android.text.style.ForegroundColorSpan;

/**
 * Clase utilitaria que se usa en varias partes de la aplicación
 */
@SuppressWarnings("all")
public final class ColorUtils {

    public static final String LS = System.getProperty("line.separator");
    public static final String LS2 = LS + LS;
    public static final float H1 = 2.2f;
    public static final float H2 = 1.7f;
    public static final float H3 = 1.4f;
    public static final float H4 = 1.1f;
    private static final ForegroundColorSpan redDefault = new ForegroundColorSpan(Color.parseColor("#A52A2A"));
    private static final ForegroundColorSpan redNightMode = new ForegroundColorSpan(Color.parseColor("#FFDAB9"));
    public static boolean isNightMode;

    public static void setNightMode(boolean nightMode) {
        //nightMode=nightMode;
    }

    public static final ForegroundColorSpan getRed() {
        return isNightMode ? redNightMode : redDefault;
    }

    public static final String getRedCode() {
        return isNightMode ? "#FFDAB9" : "#A52A2A";
    }

}

File: app/src/main/java/org/deiverbum/app/util/ColorUtils.kt
package org.deiverbum.app.util

import android.graphics.Color
import android.text.style.ForegroundColorSpan
import org.deiverbum.app.core.designsystem.theme.Red40

/**
 * Clase utilitaria para manejar la diferencia de colores en las rúbricas entre el modo noche y modo día.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
object ColorUtils {
    val LS: String? = System.lineSeparator()
    private val redDefault = ForegroundColorSpan(Color.parseColor("#A52A2A"))
    private val redNightMode = ForegroundColorSpan(Color.parseColor("#FFDAB9"))
    var isNightMode = false

    @JvmStatic
    val red: ForegroundColorSpan
        get() = if (isNightMode) redNightMode else redDefault

    @JvmStatic
    val redCode: String
        get() = if (isNightMode) "#FFDAB9" else "#A52A2A"

    var rubricColor: androidx.compose.ui.graphics.Color = Red40

}